src/odb.rs

```diff
@@ -66,6 +68,70 @@ impl<'repo> Odb<'repo> {
             Ok(())
         }
     }
+
+    /// Read object from the database.
+    pub fn read(&self, oid: Oid) -> Result<OdbObject, Error> {
+        let mut out = ptr::null_mut();
+        unsafe {
+            try_call!(raw::git_odb_read(&mut out, self.raw, oid.raw()));
+            Ok(OdbObject::from_raw(out))
+        }
+    }
+
+    /// Checks if the object database has an object.
+    pub fn exists(&self, oid: Oid) -> Result<bool, Error> {
+        unsafe { Ok(raw::git_odb_exists(self.raw, oid.raw()) != -1) }
+    }
+}
+
+/// An object from the Object Database.
+pub struct OdbObject<'a> {
+    raw: *mut raw::git_odb_object,
+    _marker: marker::PhantomData<Object<'a>>,
+}
+
+impl<'a> Binding for OdbObject<'a> {
+    type Raw = *mut raw::git_odb_object;
+
+    unsafe fn from_raw(raw: *mut raw::git_odb_object) -> OdbObject<'a> {
+        OdbObject {
+            raw: raw,
+            _marker: marker::PhantomData,
+        }
+    }
+
+    fn raw(&self) -> *mut raw::git_odb_object { self.raw }
+}
+
+impl<'a> Drop for OdbObject<'a> {
+    fn drop(&mut self) {
+        unsafe { raw::git_odb_object_free(self.raw) }
+    }
+}
+
+impl<'a> OdbObject<'a> {
+    /// Get the object type.
+    pub fn get_type(&self) -> Result<ObjectType, Error> {
```

Comment:
Similar to above (and the ones below) I don't think a `Result` needs to be returned here, right?

Reply:
I have fixed this, but it requires calling unwrap(), which feels wrong in a library, let me know what you think.